### PR TITLE
Fixed - #4844 : In peer review, deletion proposal is sent even if no reason is chosen

### LINKED ISSUE #4844: [Bug]: In peer review, deletion proposal is sent even if no reason is chosen
### Summary

In peer review, the deletion proposal is sent even if no reason is chosen.

### Steps to reproduce

1. Start peer review
2. When asked whether the picture is in scope, tap No
3. Without selecting any reason, submit

### Expected behaviour

Should not be submittable until at least a reason is chosen.

### Actual behaviour

Submitted without any reason listed. This simply shouldn't happen, as it results in a very weird (and useless) proposal.

### Device name

_No response_

### Android version

All

### Commons app version

3.1.1 master prodDebug

### Device logs

_No response_

### Screen-shots

_No response_

### Would you like to work on the issue?

Prefer not

### PR DESCRIPTION
**Pull request by arinmodi**
Description

Bug : In peer review, the deletion proposal is sent even if no reason is chosen.

Fixes #4844

What changes did you make and why?

Just add the if condition for checking the selected reasons

Tests performed

Tested build variant on ASUS_X00TD with API level 28

### CODE REVIEW

**Comment by arinmodi** [comment]
@nicolas-raoul, please review this...

**Comment by devarsh-mavani-19** [comment]
Review by student

Before 

https://user-images.githubusercontent.com/55136047/155891462-a429c97d-90d7-48da-a9b1-02a6a0191343.mp4

After

https://user-images.githubusercontent.com/55136047/155891475-b82b037f-c404-4ce5-9c1b-2ab0ccbfd0a2.mp4


Issue Solved

File: app/src/main/java/fr/free/nrw/commons/delete/DeleteHelper.java [hunk 1]
```diff
@@ -210,6 +212,9 @@ public void askReasonAndExecute(Media media,
                             reviewCallback.onFailure();
                         }
                     });
+            }else{
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/delete/DeleteHelper.java:215]: Add a comment "No reason selected"

File: app/src/main/java/fr/free/nrw/commons/delete/DeleteHelper.java [hunk 2]
```diff
@@ -186,20 +187,21 @@ public void askReasonAndExecute(Media media,
 
         alert.setPositiveButton(context.getString(R.string.ok), (dialogInterface, i) -> {
 
-            String reason = getLocalizedResources(context, Locale.ENGLISH).getString(R.string.delete_helper_ask_alert_set_positive_button_reason) + " ";
+            if(mUserReason.size() > 0){
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/delete/DeleteHelper.java:190]: You can use isEmpty method.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -691,4 +691,5 @@ Upload your first media by tapping on the add button.</string>
   <string name="feedback_sharing_data_alert">Please remove from this email any information that you are not comfortable sharing publicly. Also, please be aware that your email address with which you are posting, and the associated name and profile picture, will be visible publicly.</string>
   <string name="achievements_unavailable_beta">Achievements are only available in the prod flavor.</string>
   <string name="leaderboard_unavailable_beta">The leaderboard is only available in the prod flavor.</string>
+  <string name="select_one_reason">Please, Select At-least one reason</string>
```
Comment (4D17Y4) [inline app/src/main/res/values/strings.xml:694]: `Please select at least one reason.`

Also the name of the string could be "no_reason_selected"

**Review by 4D17Y4 — COMMENTED** [review]
Looks good to me.
Just some minor changes.  
Thanks for the student review @devarsh-mavani-19.

**Comment by arinmodi** [comment]
Hey @4D17Y4, i have done the changes

**Comment by arinmodi** [comment]
@nicolas-raoul, please review this

File: app/src/main/java/fr/free/nrw/commons/delete/DeleteHelper.java [hunk 4]
```diff
@@ -187,7 +187,7 @@ public void askReasonAndExecute(Media media,
 
         alert.setPositiveButton(context.getString(R.string.ok), (dialogInterface, i) -> {
 
-            if(mUserReason.size() > 0){
+            if(mUserReason.isEmpty()){
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/delete/DeleteHelper.java:190]: not empty ( == false ) right ?
This condition handles the non empty case.

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -691,5 +691,5 @@ Upload your first media by tapping on the add button.</string>
   <string name="feedback_sharing_data_alert">Please remove from this email any information that you are not comfortable sharing publicly. Also, please be aware that your email address with which you are posting, and the associated name and profile picture, will be visible publicly.</string>
   <string name="achievements_unavailable_beta">Achievements are only available in the prod flavor.</string>
   <string name="leaderboard_unavailable_beta">The leaderboard is only available in the prod flavor.</string>
-  <string name="select_one_reason">Please, Select At-least one reason</string>
+  <string name="no_reason_selected">Please select at least one reason</string>
```
Comment (4D17Y4) [inline app/src/main/res/values/strings.xml:694]: period at the end.

**Review by 4D17Y4 — COMMENTED** [review]
Minor fixes.

**Comment by arinmodi** [comment]
@4D17Y4 sorry i forget to add not operator, now it's fixed

**Comment by arinmodi** [comment]
Hey @4D17Y4, not able to login into beta version, it is saying timeout, what should i do ?

**Comment by 4D17Y4** [comment]
I just tried it, I can log in to beta. 
Might be an issue with your network, check your network in case that's the problem.
If you are still not able to login, it maybe a server issue. I would suggest wait for a day or two before trying again.

**Comment by 4D17Y4** [comment]
Looks good to me. 
@nicolas-raoul any final edits ?

**Comment by arinmodi** [comment]
@nicolas-raoul, please look into this...

**Comment by arinmodi** [comment]
Yes, that would be nice idea and also we can add one more thing that if the user is still pressed the ok button even if disable than we can display popUp also.

What do you think ?

**Comment by arinmodi** [comment]
Hey @nicolas-raoul, is there any callback for the changes in the list of selected reasons, we can set the button disable initially but we can't change it afterwards cause we don't have any listener for the changes in choices, what should i do ?

**Comment by arinmodi** [comment]
Hey @nicolas-raoul, please update on this, thanks

**Comment by nicolas-raoul** [comment]
> Yes, that would be nice idea and also we can add one more thing that if the user is still pressed the ok button even if disable than we can display popUp also.

Good idea!

> we can set the button disable initially but we can't change it afterwards cause we don't have any listener for the changes in choices

Not sure what is the best solution but listening sounds doable... you will have to look for code examples on the Internet and try, good luck!

**Comment by arinmodi** [comment]
Hey @nicolas-raoul, added the required changes as you idea but not implemented my idea because we can't listen for the button event once we disable it, Thanks

**Comment by 4D17Y4** [comment]
U can try `setOnCheckedChangeListener` on the checkbox. Or onclicklistener with some checks.

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
The code looks good to me!
Would you mind add unit tests?

**Comment by arinmodi** [comment]
@nicolas-raoul, Unit is already there for deleteHelper class

**Comment by arinmodi** [comment]
Hey @nicolas-raoul, added the required test, please review this

**Comment by arinmodi** [comment]
@nicolas-raoul, pushed some minor fixes to the code, please review this

File: app/src/main/java/fr/free/nrw/commons/delete/DeleteHelper.java [hunk 6]
```diff
@@ -200,20 +208,39 @@ public void askReasonAndExecute(Media media,
             String finalReason = reason;
 
             Single.defer((Callable<SingleSource<Boolean>>) () ->
-                    makeDeletion(context, media, finalReason))
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/delete/DeleteHelper.java:203]: If I understand correctly, only the indentation was changed for this block. If you actually changed code, please let me know.
That makes the code coverage check ignorable.

**Comment by arinmodi** [comment]
Yes, only indentation is changed, should i undo that ?

**Comment by nicolas-raoul** [comment]
It is usually better to address each concern in a different PR. No need to undo here though.

**Review by nicolas-raoul — APPROVED** [review]
Looks good, thanks!

**Pull request merged**